common/deliver/deliver.go

```diff
@@ -296,7 +316,7 @@ func (h *Handler) deliverBlocks(ctx context.Context, srv *Server, envelope *cb.E
 
 		iterCh := make(chan struct{})
 		go func() {
-			block, status = cursor.Next()
+			block, status = h.getNextBlock(ctx, chdr.ChannelId, cursor)
 			close(iterCh)
 		}()
 
```

Comment:
In the following select, the case ctx.Done() should be removed now. Else the newly added test could cause flake (commented on the test file to point out the particular test.